### PR TITLE
Add Pi3 internal BT support, adapt Recalbox to bluez5

### PR DESCRIPTION
**Pull request by substring**
Please make sure your PR is ready to be merged !
- [x] You added the changes in CHANGELOG.md
- [x] You choose the right repository branch to make the PR
- [x] You described the PR as below

Fixes https://github.com/recalbox/recalbox-os/issues/796

Changes :
- upgrade to bluez5 to support Pi3 internal BT
- Bluez5 is patched for some dualshock 3 clones
- Created a new sixaxispair plugin for Bluez5 that replaces sixpair bin from qtsixa
- New ps3 driver : bluez. Works perfect with genuine DS3, some shanwans, probably no GAsia. The previous drivers still exist and work fine with bluez5
- `hidd` doesn't exist in Bluez5, so there is a new demon for recalbox **btDaemon** that uses DBus
- again use DBus to connect with the `test-device`script from Bluez5
- Bluez5 stores all its data in `/var/lib/bluetooth`, which directory structure is more complex than before. So use tar to save and restore it

Please review and comment :)

### CODE REVIEW

File: configs/recalbox-x86_defconfig [hunk 1]
```diff
@@ -281,12 +281,11 @@ BR2_PACKAGE_TSLIB=y
 BR2_PACKAGE_CURL=y
 BR2_PACKAGE_BOOST_ATOMIC=y
 # BR2_PACKAGE_AVAHI_AUTOIPD is not set
-BR2_PACKAGE_BLUEZ_UTILS=y
-BR2_PACKAGE_BLUEZ_UTILS_COMPAT=y
-BR2_PACKAGE_BLUEZ_UTILS_AUDIO=y
-BR2_PACKAGE_BLUEZ_UTILS_USB=y
-BR2_PACKAGE_BLUEZ_UTILS_GATT=y
-BR2_PACKAGE_CONNMAN=y
```
Comment (paradadf) [inline configs/recalbox-x86_defconfig:289]: BR2_PACKAGE_CONNMAN=y 
Erased by mistake?

**Comment by substring** [comment]
Thanks eagle-eye @paradadf

File: package/recalbox-system/xu4/recalbox.conf [hunk 2]
```diff
@@ -104,6 +104,7 @@ controllers.bluetooth.enabled=1
 ##Enable PS3 controllers support
 controllers.ps3.enabled=1
 ## Choose an driver beetween official, shanwan and gasia if you have dualshock clones (official,shanwan,gasia)
+## or the new bluez (oficial, shanwan)
```
Comment (digitalLumberjack) [inline package/recalbox-system/xu4/recalbox.conf]:
```
## Choose a driver between bluez, official and shanwan
## bluez -> bluez 5 + kernel drivers, support official and gasia sisaxis
## official -> sixad drivers, support official and gasia sisaxis
## shanwan -> shanwan drivers, support official and shanwan sisaxis
```

File: package/recalbox-system/xu4/recalbox.conf [hunk 3]
```diff
@@ -104,6 +104,7 @@ controllers.bluetooth.enabled=1
...
 controllers.ps3.driver=official
```
Comment (digitalLumberjack) [inline package/recalbox-system/xu4/recalbox.conf]: default bluez ?

**Review by digitalLumberjack — APPROVED** [review]
Very good job

File: package/recalbox-system/xu4/recalbox.conf (hunk 3 above)
Comment (substring) [reply package/recalbox-system/xu4/recalbox.conf]: And get feedback during the unstable ? why not. I'll append a commit to the PR

File: package/recalbox-system/xu4/recalbox.conf (hunk 2 above)
Comment (substring) [reply package/recalbox-system/xu4/recalbox.conf]: rather :
`## bluez -> bluez 5 + kernel drivers, support official and shanwan sisaxis`

Comment (paradadf) [reply package/recalbox-system/xu4/recalbox.conf]: Is there any reason to have the shanwan option , if bluez handles the same?

Comment (substring) [reply package/recalbox-system/xu4/recalbox.conf]: better leave it if some people need it for now

**Comment by substring** [comment]
@digitalLumberjack looks like 8bitdo still need some tweaking, as well as ES :
- there is a small problem with the return code from the "pair" option of `recalbox-config.sh` which leads ES to believe controllers are paired everytime, which can be wrong
- the udev rules for 8bitdo's mac is case sensitive (sometimes, dunno why), have to correct the rule to lowercase

expect some commits soon

File: package/recalbox-system/xu4/recalbox.conf (hunk 2 above)
Comment (substring) [reply package/recalbox-system/xu4/recalbox.conf]: No idea of how the shanwan driver works with genuine and gasia. May work the same as the bluez driver

**Comment by substring** [comment]
One **VERY** important thing : anyone using BT will have to re-pair its BT devices when upgrading to 4.1. As said in the description of the PR, bluez5 `/var/lib/bluetooth` folder structure has nothing in common with bluez4

**Comment by digitalLumberjack** [comment]
Perfect

**Comment by Laddyacris** [comment]
error syntax : oficial
+## or the new bluez (oficial, shanwan)

-> +## or the new bluez (official, shanwan)

**Comment by paradadf** [comment]
@Laddyacris that part was changed in the last commits

**Comment by Zhan85** [comment]
How can I download this version with internal Bluetooth support?

**Comment by substring** [comment]
this is a branch, means a particular section of code. You have to compile it by yourself for now, it'sz not available as download

**Comment by Zhan85** [comment]
Got ya, thanks.  Is there anywhere I could find instructions on how to do that?

**Pull request merged**